### PR TITLE
Feature: player.triggerclient

### PR DESCRIPTION
**Pull request by cnnranderson**

### CODE REVIEW

File: server/src/TPlayer/TPlayer.cpp [hunk 1]
```diff
@@ -2026,21 +2026,16 @@ bool TPlayer::addWeapon(LevelItemType defaultWeapon)
 	if ( !settings->getBool("defaultweapons", true))
 		return false;
 
+    // TODO: (samich) - is this creating a default weapon for the server if the one of the
```
Comment (xtjoeytx) [inline server/src/TPlayer/TPlayer.cpp:2029]: for older clients (v1/v2) default weapons scripts were on the client so we just needed to acknowledge they exist. iirc, change looks good.

good catch

File: server/src/TNPC.cpp [hunk 2]
```diff
@@ -776,7 +776,6 @@ CString TNPC::setProps(CString& pProps, int clientVersion, bool pForward)
 }
 
 #ifdef V8NPCSERVER
-
```
Comment (xtjoeytx) [inline server/src/TNPC.cpp:779]: just white space changes, drop this from pr

File: server/src/Scripting/v8/V8PlayerImpl.cpp [hunk 3]
```diff
@@ -1265,6 +1268,29 @@ void Player_Function_TriggerAction(const v8::FunctionCallbackInfo<v8::Value>& ar
 	}
 }
 
+// Player Function : player.triggerclient(wep, args) -> onActionClientSide
+void Player_Function_TriggerClient(const v8::FunctionCallbackInfo<v8::Value>& args)
```
Comment (xtjoeytx) [inline server/src/Scripting/v8/V8PlayerImpl.cpp:1272]: fn signature:

player.triggerclient("gui" or "weapon", "weaponName", "args0", "arg1" ... )

source: https://forums.graalonline.com/forums/showthread.php?t=134267231

File: server/src/Scripting/v8/V8PlayerImpl.cpp [hunk 4]
```diff
@@ -1241,6 +1241,9 @@ void Player_Function_Join(const v8::FunctionCallbackInfo<v8::Value>& args)
 	}
 }
 
+// TODO (samich): Consider this deprecated based on the behavior of PLO_TRIGGERACTION.
+//                Since "clientside" is the only action the client responds to, we can
+//                rely on Player_Function_TriggerClient instead.
 void Player_Function_TriggerAction(const v8::FunctionCallbackInfo<v8::Value>& args)
 {
```
Comment (xtjoeytx) [inline server/src/Scripting/v8/V8PlayerImpl.cpp:1248]: may be possible to send a triggeraction to a player that triggers a clientside npc, ex:

player.triggeraction(player.x + 1.5, player.y, "blowup", 3);

//#CLIENTSIDE
function onActionBlowUp(params) {
  sleep(params[0]);
  doExplosion();
}

or something similar

File: server/src/Scripting/v8/V8PlayerImpl.cpp (hunk 3 above)
Comment (xtjoeytx) [reply server/src/Scripting/v8/V8PlayerImpl.cpp:1269]: just check if arg[0] is gui or weapon, pretty sure there is a possible "npc"

File: server/src/Scripting/v8/V8PlayerImpl.cpp [hunk 5]
```diff
@@ -1265,15 +1265,15 @@ void Player_Function_TriggerAction(const v8::FunctionCallbackInfo<v8::Value>& ar
...
+// Player Function : player.triggerclient("gui"/"weapon", wep, args) -> onActionClientSide
 void Player_Function_TriggerClient(const v8::FunctionCallbackInfo<v8::Value>& args)
 {
 	v8::Isolate *isolate = args.GetIsolate();
 
 	V8ENV_THROW_CONSTRUCTOR(args, isolate);
 	V8ENV_THROW_MINARGCOUNT(args, isolate, 2);
 
-	if (args[0]->IsString() && args[1]->IsString())
+	if (args[0]->IsString() && (args[0] == "gui" || args[0] == "weapon") && args[1]->IsString())
```
Comment (xx-shitai-xx) [inline server/src/Scripting/v8/V8PlayerImpl.cpp:1276]:
```suggestion
	if (args[0]->IsString() && args[1]->IsString())
```

File: server/src/Scripting/v8/V8PlayerImpl.cpp [hunk 6]
```diff
@@ -1265,15 +1265,15 @@ void Player_Function_TriggerAction(const v8::FunctionCallbackInfo<v8::Value>& ar
...
 	{
 		v8::Local<v8::Context> context = isolate->GetCurrentContext();
```
[the remark this answers is not in the document]
Comment (xx-shitai-xx) [reply server/src/Scripting/v8/V8PlayerImpl.cpp:1278]: @cnnranderson you obviously need to change following code as well

**Review by xx-shitai-xx — APPROVED** [review]

**Pull request merged**